### ISSUE TITLE
Export categories files into the project.

### ISSUE THREAD

**Issue by jamesblasco** [aws-amplify/amplify-flutter#1382]
**Describe the bug**
In the recent update there is a huge breaking change that is categories classes have been hidden and they are not public anymore and can't be used in projects using this dependency. This has had a great impact in our project as we were passing the class as a parameter in the constructor of our repositories so we could mock it.

```dart
class MockAPICategory extends Mock implements APICategory {}

class ExampleRepository {
  ExampleRepository({required APICategory apiCategory})
      : _apiCategory = apiCategory;

  final APICategory _apiCategory;
   
  /// Example query
  Future<String?> query({required ApiRequest request}) async {
    final operation = _apiCategory.query(request: request);

    final response = await operation.response;

    if (response.errors.isNotEmpty) {
      throw GraphQLOperationException(response.errors);
    }
    return response.data!;
  }
}
```

**What caused this change**

The change was to move 
lib/categories/amplify_categories.dart
to
lib/src/categories/amplify_categories.dart

Anything inside src is considered private for the analyzer following the official convention 

<img width="938" alt="Screenshot 2022-02-21 at 11 24 14" src="https://user-images.githubusercontent.com/19904063/154936382-a0edccf1-8217-467f-aa1b-85f6f6117f98.png">

https://dart.dev/guides/libraries/create-library-packages

**Proposal**

Export the categories via another dart file in the lib folder

-lib
  - src/
     - categories/amplify_categories.dart' (private)
  - amplify_flutter.dart (public)
  - amplify_categories (public)

So when needed to use the APICategory class we would do
- import 'package:amplify_flutter/amplify_categories.dart';

**Comment by dnys1** [comment]
Hi @jamesblasco - so sorry for this oversight on our part. This was not intentional. We will release a fix for this soon, and in the meantime, feel free to import these directly:

```dart
import 'package:amplify_flutter/src/categories/amplify_categories.dart';
```

**Comment by ragingsquirrel3** [comment]
This has been fixed with 0.4.1 release.
